src/openssl/lib_crypto.cr

```diff
@@ -1,8 +1,17 @@
 {% begin %}
   lib LibCrypto
-    OPENSSL_111 = {{ `command -v pkg-config > /dev/null && pkg-config --atleast-version=1.1.1 libcrypto || printf %s false`.stringify != "false" }}
-    OPENSSL_110 = {{ `command -v pkg-config > /dev/null && pkg-config --atleast-version=1.1.0 libcrypto || printf %s false`.stringify != "false" }}
-    OPENSSL_102 = {{ `command -v pkg-config > /dev/null && pkg-config --atleast-version=1.0.2 libcrypto || printf %s false`.stringify != "false" }}
+    {% from_libressl = (`hash pkg-config 2> /dev/null || printf %s false` != "false") &&
+                       (`test -f $(pkg-config --silence-errors --variable=includedir libcrypto)/openssl/opensslv.h || printf %s false` != "false") &&
+                       (`printf %s "#include <openssl/opensslv.h>\nlibressl_version_number" | #{(env("cc") || "cc").id} #{`pkg-config --cflags --silence-errors libcrypto || true`.chomp}  -E -`.chomp.split('\n').last == "libressl_version_number") %}
```

Comment:
perhaps use `$(pkg-config ...)` here instead of ``#{`pkg-config ...`}``

Reply:
`#{(env("cc") || "cc").id}` could be inserted directly into the command, as well.